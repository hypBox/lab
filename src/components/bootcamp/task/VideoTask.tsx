import React, { FunctionComponent } from "react"
import { IVideoTask } from "../../../types"

export interface VideoTaskProps {
  task: IVideoTask
}

const VideoTask: FunctionComponent<VideoTaskProps> = ({ task }) => {
  return <h2>I am a video task titled {task.title}</h2>
}

export default VideoTask
