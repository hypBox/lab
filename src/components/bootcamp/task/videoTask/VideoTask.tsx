import React, { FunctionComponent } from "react"
import { IVideoTask } from "../../../../types"
import ReactPlayer from "react-player"

export interface VideoTaskProps {
  task: IVideoTask
}

const VideoTask: FunctionComponent<VideoTaskProps> = ({ task }) => {
  return (
    <div className="player-wrapper">
      <ReactPlayer
        className="react-player"
        url={task.url}
        controls={true}
        width="100%"
        height="100%"
      />
    </div>
  )
}

export default VideoTask
