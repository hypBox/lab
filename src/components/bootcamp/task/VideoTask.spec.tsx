import React from "react"
import { render } from "@testing-library/react"
import VideoTask, { VideoTaskProps } from "./VideoTask"

jest.mock("react-player", () => {
  const ReactPlayer = (props: any) => {
    props.onProgress({ played: 100 })
    return <div></div>
  }
  return ReactPlayer
})

const props: VideoTaskProps = {
  onProgress: jest.fn(),
  task: {
    id: "t1",
    title: "Bootcamp Overview",
    type: "video",
    url: "",
    length: "1m 24s",
    progress: 100,
  },
}

const renderComponent = () => render(<VideoTask {...props} />)

describe("Video Task Component", () => {
  it("renders as expected", () => {
    const { container } = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })
})
