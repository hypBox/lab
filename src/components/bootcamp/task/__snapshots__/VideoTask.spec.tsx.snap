// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Task Component renders as expected 1`] = `
<div
  class="makeStyles-playerWrapper-1"
>
  <div />
</div>
`;
